### DISCUSSION TITLE
Fitment Alignment Graphic Guide Aid for Straight/Correct Installation of OLED 0.96' 128x64 Display

### DISCUSSION THREAD

**Discussion by SynthfulDwarphus** [keirf/flashfloppy#669] [Ideas]
Hello all,

I propose addition to firmware of a single-screen graphic image to serve as an aid/guide for fitment/alignment of an OLED 0.96" 128x64 straight/correctly in the Gotek housing standard factory display window.

Such a graphic alignment guide will be beneficial because it's difficult to gauge fitment of the .96 OLED in the housing window due to tight tolerances and difficulty in seeing entire lit display area while mounting it (the "-inverse" display CFG command can help but only goes so far, plus a graphic aid serves two additional purposes - see my separate 20-character and "-xHLines" Idea discussion posts). 

It would be great to be able to make the alignment graphic immediately visible directly with a FlashFloppy front panel menu command, and then for the graphic to remain visible indefinitely until any button is pressed/the encoder turned.

The graphic should have these three features:

1 ) A single-pixel graphic line running he entire perimeter of the OLED's display area (all 4 sides), so during mounting you can check for both straight alignment and that the entire display area is viewable within the housing window (remember, .96 OLED has very tight tolerances).

2 )  Three vertical dashed-pattern lines from top to bottom of display denoting the extreme right-most edge of 6x13 font modes for 16-character, and 18-character, and (separately suggested idea post) 20-character widths.

3 ) Two horizontal dashed-pattern lines across top of display at minus-4 and minus-8 horizonal lines reduced/truncated display height positions (and additional positions, if needed), to correspond with my separate idea post display height reduction "-xHLines" command.

ATTACHED TO THIS POST is a mock-up graphic PNG file of the alignment aid that I created in actual 128x64-pixel size (I used the draw tool at pixilart.com). It needs a couple of basic improvements, but conveys the gist.

I am not a programmer, so I'm not able to implement the alignment graphic aid feature in the FlashFloppy firmware myself, and therefore kindly request the abilities of someone else that is able to do so.

I think ths feature would be beneficial to many users, and, of course, I'm completely open to improvements/suggestions!

Thanks,
SynD
![FlashFloppy Fitment Guide Graphic for 0 96 OLED 128x64 v2](https://user-images.githubusercontent.com/105606636/180672210-5a921df6-6b4d-43ab-9eb3-746016319727.png)

**Comment by keirf** [comment]
Did you know there is now a factory model with 0.96 128x64 OLED and rotary encoder already fitted?
eg https://www.ebay.co.uk/itm/144279447370

**Comment by SynthfulDwarphus** [comment]
Yes, thanks, but the pre-modded Gotek drive is about three times the price
as buying the basic Gotek plus mod parts, and I'm looking at needing
another 10-15+ drives, so I need to stick with modding myself,
unfortunately.

I'm sure plenty of other users are under similar price constraints, and
would appreciate the simple but effective fitment guide graphic feature
when they mod their FF Gotek's with 0.96 OLEDs, too!

Regards.

**Reply by keirf** [reply]
At that quantity perhaps buy from factory. Contact Yan at guiie@163.com

**Comment by SynthfulDwarphus** [comment]
Thanks for the tip, I wasn't aware that was an option!

And regarding the fitment graphic addition to firmware, is that not
possible, then?
(I have a few I already bought to mod yet , plus I believe this would help
others!)

Cheers.

**Reply by keirf** [reply]
Why not just configure to print image name on all rows (using display-order= option) and select an image called say XXXXXXXXXXXXXXXXXXXXXXXXXXXX.IMG

This would fill the display with Xs and you can centre as you wish. What's the need for a graphic.

**Reply by keirf** [reply]
Also if you buy from factory ask for 435 based Goteks. New chip with more RAM. Worth the extra dollar.

**Reply by SynthfulDwarphus** [reply]
Right, I mentioned in post that "-inverse" display CFG fills screen with positive pixels (possibly better than all "x"s), but:

Single pixel line only along perimeter assists both straight alignment and over-mounting (single pixel disappears if over-mounted), and

Dashed lines assist anyone mounting display right-ward in close proximity to 16/18/(20) characters, and

-4/-8 dashed line(s) assist alignment & over-mounting if "-xHlines" feature implemented.

Thank you.

**Comment by SynthfulDwarphus** [comment]
Definitely will do so, thanks!

**Comment by SynthfulDwarphus** [comment]
Right, I mentioned in post that "-inverse" display CFG fills screen with
positive pixels (possibly better than all "x"s), but:

1) Single pixel line only along perimeter assists both straight alignment
and over-mounting (single pixel disappears if over-mounted), and

2) Dashed lines assist anyone mounting display right-ward in close
proximity to 16/18/(20) characters, and

3) -4/-8 dashed line(s) assist alignment & over-mounting if "-xHlines"
feature implemented.

Thank you.
